include/matx/operators/outer.h

```diff
@@ -0,0 +1,160 @@
+////////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// COpBright (c) 2021, NVIDIA Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above cOpBright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above cOpBright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the cOpBright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COpBRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COpBRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+
+#include "matx/core/type_utils.h"
+#include "matx/operators/base_operator.h"
+#include "matx/transforms/outer.h"
+
+namespace matx
+{
+  namespace detail {
+    template <typename OpA, typename OpB>
+    class OuterOp : public BaseOp<OuterOp<OpA, OpB>>
+    {
+      private:
+        OpA a_;
+        OpB b_;
+        float alpha_;
+        float beta_;
+        static constexpr int RANK = remove_cvref_t<OpB>::Rank();
+        std::array<index_t, RANK> out_dims_;
+        mutable matx::tensor_t<typename OpA::scalar_type, RANK> tmp_out_;
+
+      public:
+        using matxop = bool;
+        using scalar_type = typename OpA::scalar_type;
+        using matx_transform_op = bool;
+        using outer_xform_op = bool;
+
+        __MATX_INLINE__ std::string str() const { 
+          return "outer(" + get_type_str(a_) + "," + get_type_str(b_)  + ")";
+        }
+
+        __MATX_INLINE__ OuterOp(const OpA &A, const OpB &B, float alpha, float beta) : 
+              a_(A), b_(B), alpha_(alpha), beta_(beta) {
+          
+          for (int r = 0; r < Rank(); r++) {
```

Comment:
is this correct?  I would expect an outer product to increase the rank by 1?   ie:  matrix = outer(vec,vec)

Reply:
Fixed using max